src/rich-text-editor.component.js

```diff
@@ -72,29 +73,31 @@ export const RichTextEditor = forwardRef((props, editorRef) => {
         clearTimeout(timeout)
       }
     } 
-  }, [focused])
+  }, [isFocused()])
 
   useEffect(() => {
     richTextContext.selectRangeFromBeforeBlur = () => {
       if (divRef.current && document.activeElement !== divRef.current && !divRef.current.contains(document.activeElement)) {
-        if (selectionRangeBeforeBlurRef.current) {
-          const currentSelection = window.getSelection()
-          currentSelection.removeAllRanges()
-          currentSelection.addRange(selectionRangeBeforeBlurRef.current)
-        } else {
-          divRef.current.focus()
-        }
+        divRef.current.focus()
```

Comment:
calling focus just so happens to select the previously selected text from before the blur. So we actually don't need to call removeAllRanges and addRange with the last range.

Reply:
nice!